src/benchmarklib/benchmark_runner.cpp

```diff
@@ -0,0 +1,449 @@
+
+  // If only one file was specified, add it and return
+  if (fs::is_regular_file(path)) {
+    if (is_table_file(table_path)) {
+      tables.push_back(table_path);
+      return tables;
+    } else {
+      throw std::runtime_error("Specified file '" + table_path + "' is not a .csv or .tbl file");
+    }
+  }
+
+  // Recursively walk through the specified directory and add all files on the way
+  for (const auto& entry : fs::recursive_directory_iterator(path)) {
+    const auto filename = entry.path().string();
+    if (fs::is_regular_file(entry) && is_table_file(filename)) {
+      tables.push_back(filename);
+    }
+  }
+
+  return tables;
+}
+
+NamedQueries BenchmarkRunner::_parse_query_path(const std::string& query_path) {
+  const auto is_sql_file = [](const std::string& filename) { return boost::algorithm::ends_with(filename, ".sql"); };
+
+  fs::path path{query_path};
+  if (!fs::exists(path)) {
+    throw std::runtime_error("No such file or directory '" + query_path + "'");
+  }
+
+  if (fs::is_regular_file(path)) {
+    if (is_sql_file(query_path)) {
+      return _parse_query_file(query_path);
+    } else {
+      throw std::runtime_error("Specified file '" + query_path + "' is not an .sql file");
+    }
+  }
+
+  // Recursively walk through the specified directory and add all files on the way
+  NamedQueries queries;
+  for (const auto& entry : fs::recursive_directory_iterator(path)) {
+    const auto filename = entry.path().string();
+    if (fs::is_regular_file(entry) && is_sql_file(filename)) {
+      const auto file_queries = _parse_query_file(filename);
+      queries.insert(queries.end(), file_queries.begin(), file_queries.end());
+    }
+  }
+
+  return queries;
+}
+
+NamedQueries BenchmarkRunner::_parse_query_file(const std::string& query_path) {
+  auto query_id = 0u;
+
+  std::ifstream file(query_path);
+  const auto filename = fs::path{query_path}.stem().string();
+
+  NamedQueries queries;
+  std::string query;
+  while (std::getline(file, query)) {
+    if (query.empty() || query.substr(0, 2) == "--") {
+      continue;
+    }
+
+    const auto query_name = filename + '.' + std::to_string(query_id);
+    queries.emplace_back(query_name, std::move(query));
+    query_id++;
+  }
+
+  return queries;
+}
+
+cxxopts::Options BenchmarkRunner::get_default_cli_options(const std::string& benchmark_name) {
+  cxxopts::Options cli_options{benchmark_name};
+
+  // clang-format off
+  cli_options.add_options()
+    ("help", "print this help message")
+    ("v,verbose", "Print log messages", cxxopts::value<bool>()->default_value("false"))
+    ("r,runs", "Maximum number of runs of a single query", cxxopts::value<size_t>()->default_value("1000")) // NOLINT
+    ("c,chunk_size", "ChunkSize, default is 2^32-1", cxxopts::value<ChunkOffset>()->default_value(std::to_string(Chunk::MAX_SIZE))) // NOLINT
+    ("t,time", "Maximum seconds within which a new query(set) is initiated", cxxopts::value<size_t>()->default_value("5")) // NOLINT
```

Comment:
I don't understand this

Reply:
I would guess this is the maximum runtime for a benchmark of a query. Then I would suggest 'Maximum seconds a single query will be benchmarked'

Reply:
again, copied so i'm not 100% certain (maybe @mrzzzrm can clarify)

there are two modes, one that benchmarks all queries `min(runs, time)` times and one that takes a set of queries and runs them in one block (kind of as if they were one query) with the same  `min(runs, time)` constraint.

so after `time` is up, either a new query is run or the queryset that was run is over, depending on the mode.

but i agree, this isn't clear. i think @nilsthamm's suggestion is better and i will adapt it to something like that